Sources/Sentry/SentryThreadInspector.m

```diff
@@ -40,15 +77,82 @@ - (id)initWithStacktraceBuilder:(SentryStacktraceBuilder *)stacktraceBuilder
         sentryThread.name = [self getThreadName:thread];
 
         sentryThread.crashed = @NO;
-        bool isCurrent = thread == sentrycrashthread_self();
+        bool isCurrent = thread == currentThread;
         sentryThread.current = @(isCurrent);
 
-        // For now we can only retrieve the stack trace of the current thread.
         if (isCurrent) {
             sentryThread.stacktrace = [self.stacktraceBuilder buildStacktraceForCurrentThread];
         }
 
-        [threads addObject:sentryThread];
+        // We need to make sure the main thread is always the first thread in the result
+        if ([self.machineContextWrapper isMainThread:thread])
+            [threads insertObject:sentryThread atIndex:0];
+        else
+            [threads addObject:sentryThread];
+    }
+
+    return threads;
+}
+
+/*
+ We are not sharing code with 'getCurrentThreads' because both methods use different approaches.
```

Comment:
`m`: I think we just need to mention that using the above approach can lead to crashes and deadlocks and we must not call ObjC when the threads are suspended.